Milestone 8. Add basic Javascript functionality to menu

indigodavid opened the pull request:
This milestone modifies several parts of the project, as follows:

- Adds linters for Javascript
- Adds new index.js file for Javascript implementation
- Modifies index.html to add the newly created script
- Modifies menu icon behavior so that it works with a background image
instead of img tag
- Adds close.svg for the menu icon when active
- Adds new classes in CSS files to handle the new functionalities

indigodavid committed:
Add basic Javascript functionality to menu

This commit modifies several parts of the project, as follows:

- Adds linters for Javascript
- Adds new index.js file for Javascript implementation
- Modifies index.html to add the newly created script
- Modifies menu icon behavior so that it works with a background image
instead of img tag
- Adds close.svg for the menu icon when active
- Adds new classes in CSS files to handle the new functionalities

indigodavid committed:
Modify appearance of mobile menu

indigodavid committed:
Edit index.js to comply ESLint

nicupop729 reviewed (changes requested):
Hi @indigodavid, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

<img src="https://media3.giphy.com/media/cKKXNlTYino7hWNXwl/giphy.gif"/>

You have submitted a milestone, that was supposed to be built in a pair-programming activity.
Kindly, invite your partner to work together following the driver-navigator approach so that both of you commit to this milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

Andyveloper committed:
Add transition to menu & interactive menu links

mntedev committed:
update display option for links

indigodavid committed:
Fix linters errors and animation

zieeco commented on css/style.css:
```diff
@@ -455,3 +458,53 @@ form > textarea {
+    background-image: url(../images/close.svg);
+  }
+
+  #navigation-links.nav-active {
+    position: absolute;
+    display: flex;
+    transition: transform 0.2s;
+    transform: translateX(0);
+    flex-direction: column;
+    align-content: center;
+    inset: 92px 0;
+    height: 90vh;
+    background-color: var(--color-menu);
+  }
```
**[REQUIRED CHANGES]**

- Please, kindly adjust your styling so that your mobile menu have a width of `100%` and height of `100%` so that a user must click on the menu links before seeing the rest content in your page because right now, a user can ignore clicking the menu list items and just scroll down to view the contents of your page. 
Please, kindly fix this..

zieeco replied:
**[REQUIRED CHANGES]**

- Also, Please add the white background and the black line to match the figma design. This could be achieved by placing your menu in a parent container so the parent has the white background and the black line. The menu which the child now takes up 90% height and 100% width of it parent container.

indigodavid replied:
> **[REQUIRED CHANGES]**
> 
> * Also, Please add the white background and the black line to match the figma design. This could be achieved by placing your menu in a parent container so the parent has the white background and the black line. The menu which the child now takes up 90% height and 100% width of it parent container.

Hello @zieeco, thanks for your review. I don't fully understand your second requirement, kindly provide a screenshot of the white background and black line you want me to edit, so I can make the changes.

zieeco reviewed (changes requested):
Hi @Team


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._


<img src="https://media0.giphy.com/media/8UGVQESpDMMQl7yd3F/giphy.gif"/>


## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

indigodavid committed:
Edit javascript menu

Change styling of menu to fixed to prevent scrolling when open
Add white-bar element and black bar

Wes-Isaac reviewed (approved):
Hi @indigodavid & @worashf, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
